src/audio/src/src.c

```diff
@@ -335,57 +335,64 @@ int src_polyphase_init(struct polyphase_src *src, struct src_param *p,
 }
 
 /* Fallback function */
-static void src_fallback(struct comp_dev *dev, struct comp_data *cd,
-			 const struct audio_stream __sparse_cache *source,
-			 struct audio_stream __sparse_cache *sink, int *n_read, int *n_written)
+static int src_fallback(struct comp_data *cd,
+			struct source __sparse_cache *source, struct sink __sparse_cache *sink)
 {
-	*n_read = 0;
-	*n_written = 0;
+	return 0;
 }
 
 /* Normal 2 stage SRC */
-static void src_2s(struct comp_dev *dev, struct comp_data *cd,
-		   const struct audio_stream __sparse_cache *source,
-		   struct audio_stream __sparse_cache *sink, int *n_read, int *n_written)
+static int src_2s(struct comp_data *cd,
+		  struct source __sparse_cache *source, struct sink __sparse_cache *sink)
 {
 	struct src_stage_prm s1;
 	struct src_stage_prm s2;
 	int s1_blk_in;
 	int s1_blk_out;
 	int s2_blk_in;
 	int s2_blk_out;
-	void *sbuf_addr = cd->delay_lines;
+	uint32_t n_read = 0, n_written = 0;
+	int ret;
+	void __sparse_cache *buffer_start;
 	void *sbuf_end_addr = &cd->delay_lines[cd->param.sbuf_length];
 	size_t sbuf_size = cd->param.sbuf_length * sizeof(int32_t);
-	int nch = audio_stream_get_channels(source);
+	int nch = source_get_channels(source);	/* src channels must == sink channels */
```

Comment:
this comment looks like it belongs in prepare

Reply:
Disagree. It is usage of new interface, the very last commit

Reply:
yes but we should not be making comments about our assumed state in the processing function, we should be validating that in prepare. It just feels weird the way its written like a check is assumed here. Whereas I see its used to explain the logic and why there is an optimization. Maybe "chan sink == chan src therefore we only need to use one"